pkg/kubelet/cm/topologymanager/policy_strict.go

```diff
@@ -69,3 +72,97 @@ func (p *strictPolicy) Merge(permutation []TopologyHint) TopologyHint {
 	// preferred allocation was used to generate the affinity mask or not.
 	return TopologyHint{mergedAffinity, preferred}
 }
+
+func (p *strictPolicy) Merge(providersHints []map[string][]TopologyHint) TopologyHint {
+	// Set the default hint to return from this function as an any-socket
+	// affinity with an unpreferred allocation. This will only be returned if
+	// no better hint can be found when merging hints from each hint provider.
+	defaultAffinity, _ := socketmask.NewSocketMask()
+	defaultAffinity.Fill()
+	defaultHint := TopologyHint{defaultAffinity, false}
+
+	// Loop through all provider hints and save an accumulated list of the
+	// hints returned by each hint provider. If no hints are provided, assume
+	// that provider has no preference for topology-aware allocation.
+	var allProviderHints [][]TopologyHint
```

Comment:
Now that policy is in charge of selecting the best hint we can probably improve this by only appending hints with a single bit set in the socketmask as any permutation containing a hint with more than one bit set will lead to preferred=false in the selected bestHint.